### PR TITLE
[REVIEW] Align memory resource allocation sizes to 8-byte

### PR DESCRIPTION
**Pull request by OlivierNV**
Align size of memory allocations to 8 bytes.

### CODE REVIEW

File: include/rmm/mr/device_memory_resource.hpp [hunk 1]
```diff
@@ -23,6 +23,15 @@ using cudaStream_t = struct CUstream_st*;
 
 namespace rmm {
 namespace mr {
+
+/**
+ * Align up to a power of 2, align_bytes is expected to be a nonzero power of 2
+ */
+inline std::size_t align_up(std::size_t v, std::size_t align_bytes) noexcept
```
Comment (harrism) [inline include/rmm/mr/device_memory_resource.hpp]: Suggest moving this up to an anonymous namespace.

Comment (jrhemstad) [reply include/rmm/mr/device_memory_resource.hpp]: Should be `detail`. Don't want an anonymous namespace in a header file.

Comment (OlivierNV) [reply include/rmm/mr/device_memory_resource.hpp]: Any suggestion for the fiule this should go in ? There doesn't appear to be a detail namespace, and include/rmm/detail only appears to contain memory_manager.hpp, which includes device_memory_resource.hpp

Comment (harrism) [reply include/rmm/mr/device_memory_resource.hpp]: Jake just means keep it in this file, but add a `namespace detail` scope around this function.

Comment (harrism) [reply include/rmm/mr/device_memory_resource.hpp]: At least I assume that's what he means. :)

Comment (OlivierNV) [reply include/rmm/mr/device_memory_resource.hpp]: Sounds reasonable ~ updated the file.

File: include/rmm/mr/device_memory_resource.hpp [hunk 2]
```diff
@@ -22,6 +22,18 @@
 using cudaStream_t = struct CUstream_st*;
 
 namespace rmm {
+
+namespace detail {
+/**
+ * Align up to a power of 2, align_bytes is expected to be a nonzero power of 2
+ */
+inline std::size_t align_up(std::size_t v, std::size_t align_bytes) noexcept
```
Comment (jrhemstad) [inline include/rmm/mr/device_memory_resource.hpp]: This should be documented consistent with https://github.com/rapidsai/cudf/blob/branch-0.11/cpp/src/examples/documentation_example.cpp

Comment (OlivierNV) [reply include/rmm/mr/device_memory_resource.hpp]: Updated. Couldn't find the eyeroll emoji in in github.

**Pull request merged**